### PR TITLE
Program Once-Over

### PR DESCRIPTION
**Pull request by buffalojoec**
Some small changes to finish up the Funnel program.

### CODE REVIEW

File: program/src/processor.rs [hunk 1]
```diff
@@ -73,11 +73,17 @@ fn process_distribute_rewards(
...
+    // Ensure the provided treasury account has the correct address derived
+    // from the stake config account.
+    if find_treasury_pda(stake_config_info.key).0 != *treasury_info.key {
         return Err(PaladinFunnelError::IncorrectTreasuryAddress.into());
     }
 
+    // Ensure the provided treasury account is owned by the governance program.
+    if treasury_info.owner != &GOVERNANCE_PROGRAM_ID {
+        return Err(ProgramError::InvalidAccountOwner);
+    }
```
Comment (joncinque) [inline program/src/processor.rs:85]: Unless I missed something, the treasury shouldn't be owned by the governance

**Review by joncinque — COMMENTED** [review]
Just one question, looks good otherwise!

File: program/src/processor.rs (hunk 1 above)
Comment (buffalojoec) [reply program/src/processor.rs:85]: Bahh I forgot we decided to make it a system account for `system_instruction::transfer`. Good catch.

Comment (buffalojoec) [reply program/src/processor.rs]: I shed that first commit from this PR.

**Review by joncinque — APPROVED** [review]

**Pull request merged**